polyaxon_client/tracking/base.py

```diff
@@ -24,7 +24,6 @@
 class BaseTracker(object):
     REQUIRES_OUTPUTS = True
 
-    @check_no_op
```

Comment:
I think this will be confusing for users, if I set `POLYAXON_NO_OP`, I should expect the behaviour to be semantically consistent.

I would suggest to handle the logic with a different flag `POLYAXON_LOCAL`, or better `POLYAXON_OFFLINE` which will be the default mode when the user is running a hyper params tuning. This way we can turn on/off a behaviour for that specific case, similar to `POLYAXON_IS_MANAGED` or `POLYAXON_IN_CLUSTER`.

The offline should actually handle the same behaviour but instead of logging stuff to the API, it should log it to local disk.

For this PR, since the main objective is to resolve the env vars to the users (data, outputs, params), I am not sure if the easiest way is to create an offline client that just mocks the behaviour of a normal client?

Reply:
Yes, introducing new env variable won't break existing behaviour.

Offline client is handy to localize changes in a single place.

Reply:
@mouradmourafiq do you think that local mode should be always offline?

Reply:
Not necessarly, and I see that using `POLAYAXON_LOCAL` as flag could be confusing, because you can use the local run and do full tracking of the experiment similar to some paid SaaS products, where users run their xps on their computers, but do the tracking online. Another use case is when the GPUs of the cluster are all used, and the user has local gpu that she wants to use without losing the progress of her work.

So I think the local mode, should not be any different, the client should just respect the `POLYAXON_NO_OP` to do no tracking and ignore all polyaxon operations, and `POLYAXON_OFFLINE` to do other stuff, users can syng the results after.

As a start, I would suggest to do the minimal work possible, i.e. `POLYAXON_OFFLINE` can just do things you need now (exposing env vars) and ignore all other operations, I image that in that case for some function calls we can just start by adding a new decorator to ignore those calls, similar to `@no_op`, and calls that are need will be executed because they will only respect the `no_op` decorator.

For instance, all functions that do client calls, can be further decorated with `check_offline`, and the functions that you need, e.g. `get_outputs_path` will not be decorated. The `check_offline` should just behave like no op for the the offline flag.

Reply:
@mouradmourafiq I've updated the PR, introduced `check_offline` decorated and settings.IS_OFFLINE flag.